Sources/System/IOCompletion.swift

```diff
@@ -0,0 +1,56 @@
+@_implementationOnly import CSystem
+
+public struct IOCompletion: ~Copyable {
+    let rawValue: io_uring_cqe
+}
+
+extension IOCompletion {
+    public struct Flags: OptionSet, Hashable, Codable {
+        public let rawValue: UInt32
+
+        public init(rawValue: UInt32) {
+            self.rawValue = rawValue
+        }
+        
+        public static let allocatedBuffer = Flags(rawValue: 1 << 0)
```

Comment:
Would be great to have a link to e.g.

https://github.com/axboe/liburing/blob/92e858ad3f23ab2cdacbb5123ddce4c437452d9f/src/include/liburing/io_uring.h#L446

so one can see where these magic values comes from.